ports/esp32/mpconfigport.h

```diff
@@ -274,15 +276,25 @@
 #define MICROPY_HW_USB_CDC                  (MICROPY_HW_ENABLE_USBDEV)
 #endif
 
+// Enable stdio over native USB peripheral HID via TinyUSB
```

Comment:
This feature is unrelated to this PR, but also it's probably not a feature we will add to MicroPython: we already have the [machine.USBDevice](https://docs.micropython.org/en/latest/library/machine.USBDevice.html) interface which allows creating any custom USB HID device with Python code, instead.

(Suggest removing it from this PR, at least. You can submit it in another PR if you want, but note that it probably isn't something we want to support.)